packages/compiler/src/server/serverlib.ts

```diff
@@ -860,14 +861,34 @@ export function createServer(host: ServerHost): Server {
       const tspDiag = currentDiagnosticIndex.get(vsDiag.data?.id);
       if (tspDiag === undefined || tspDiag.codefixes === undefined) continue;
 
+      if (
+        vsDiag.codeDescription?.href &&
+        tspDiag.codefixes?.findIndex((x) => x.id === "show-linter-rule-doc-url") === -1
+      ) {
+        mutate(tspDiag.codefixes).push(
+          defineCodeFix({
+            id: "show-linter-rule-doc-url",
+            label: `Open document`,
+            fix: (context) => {},
+          }),
+        );
+      }
+
       for (const fix of tspDiag.codefixes ?? []) {
+        const currentCmd =
+          fix.id === "show-linter-rule-doc-url" ? Commands.OPEN_RULE_DOC : Commands.APPLY_CODE_FIX;
```

Comment:
it's better to if-else the fix.id with their own logic to make the code more readable and also makes it easier to extend in the future